electron-app/src/server/websites/cohost/cohost.service.ts

```diff
@@ -0,0 +1,305 @@
+import { Injectable } from '@nestjs/common';
+import {
+  DefaultOptions,
+  FileRecord,
+  FileSubmission,
+  FileSubmissionType,
+  CohostFileOptions,
+  CohostNotificationOptions,
+  PostResponse,
+  Submission,
+  SubmissionPart,
+  SubmissionRating,
+} from 'postybirb-commons';
+import UserAccountEntity from 'src/server/account/models/user-account.entity';
+import { PlaintextParser } from 'src/server/description-parsing/plaintext/plaintext.parser';
+import Http from 'src/server/http/http.util';
+import { CancellationToken } from 'src/server/submission/post/cancellation/cancellation-token';
+import {
+  FilePostData,
+  PostFile,
+  PostFileRecord,
+} from 'src/server/submission/post/interfaces/file-post-data.interface';
+import { PostData } from 'src/server/submission/post/interfaces/post-data.interface';
+import { ValidationParts } from 'src/server/submission/validator/interfaces/validation-parts.interface';
+import FileSize from 'src/server/utils/filesize.util';
+import FormContent from 'src/server/utils/form-content.util';
+import WebsiteValidator from 'src/server/utils/website-validator.util';
+import ImageManipulator from 'src/server/file-manipulation/manipulators/image.manipulator';
+import { LoginResponse } from '../interfaces/login-response.interface';
+import { ScalingOptions } from '../interfaces/scaling-options.interface';
+import { Website } from '../website.base';
+import _ from 'lodash';
+import cheerio from 'cheerio';
+import { CohostFileOptionsEntity } from 'postybirb-commons/lib/websites/cohost/cohost.file.options';
+
+@Injectable()
+export class Cohost extends Website {
+  BASE_URL: string = 'https://cohost.org';
+  readonly MAX_CHARS: number = 5000;
+  acceptsFiles: string[] = ['jpeg', 'jpg', 'jfif', 'png', 'pjpeg', 'pjp', 'gif', 'webp', 'svg', 'aac', 'm4a', 'm4b', 'flac', 'mp3', 'mpega', 'mp2', 'wav'];
+  acceptsAdditionalFiles: boolean = true;
+
+  page_name: string = "";
+  cookie: any;
+  cohostPlus: boolean;
+
+  readonly defaultDescriptionParser = PlaintextParser.parse;
+  readonly usernameShortcuts = [
+    {
+      key: 'co',
+      url: 'https://cohost.org/$1',
+    },
+  ];
+
+  async checkLoginStatus(data: UserAccountEntity): Promise<LoginResponse> {
+    const status: LoginResponse = { loggedIn: false, username: null };
+    const res = await Http.get<string>(`${this.BASE_URL}/rc/user/settings`, data._id);
+    this.cookie = (await Http.getWebsiteCookies(data._id, this.BASE_URL))[0];
+    this.cookie = `${this.cookie.name}=${this.cookie.value}`;
+    
+    if (res.body.includes('change password')) {
+      status.loggedIn = true;
+      const $ = cheerio.load(res.body);
+      status.username = $('input[type=email]').attr('value');
+      this.page_name = $("*[id*=headlessui] img[alt]").attr("alt");
```

Comment:
Moved pageName to AccountInformation. I couldn't figure out how to implement hasCohostPlus into it though, since it was required in the getScaleOptions and validateFileSubmission methods, where I can't easily get the accountId. I just removed it for now and kept the limit at 5MB instead of increasing it to 10MB for Plus users.

Reply:
Yeah its just a limit to the current implementation right now. Keep forgetting to add the parameter id to that function.